x-pack/plugins/timelines/server/search_strategy/timeline/index.ts

```diff
@@ -5,48 +5,165 @@
  * 2.0.
  */
 
-import { map, mergeMap } from 'rxjs/operators';
+import { map, mergeMap, catchError } from 'rxjs/operators';
+import { from } from 'rxjs';
+import {
+  isValidFeatureId,
+  mapConsumerToIndexName,
+  ALERTS_CONSUMERS,
+} from '@kbn/rule-data-utils/target/alerts_as_data_rbac';
+
+import {
+  AlertingAuthorizationEntity,
+  AlertingAuthorizationFilterType,
+  PluginStartContract as AlertPluginStartContract,
+} from '../../../../alerting/server';
 import {
   ISearchStrategy,
   PluginStart,
+  SearchStrategyDependencies,
   shimHitsTotal,
 } from '../../../../../../src/plugins/data/server';
-import { ENHANCED_ES_SEARCH_STRATEGY } from '../../../../../../src/plugins/data/common';
 import {
   TimelineFactoryQueryTypes,
   TimelineStrategyResponseType,
   TimelineStrategyRequestType,
+  EntityType,
 } from '../../../common/search_strategy/timeline';
 import { timelineFactory } from './factory';
 import { TimelineFactory } from './factory/types';
+import {
+  ENHANCED_ES_SEARCH_STRATEGY,
+  ISearchOptions,
+} from '../../../../../../src/plugins/data/common';
 
 export const timelineSearchStrategyProvider = <T extends TimelineFactoryQueryTypes>(
-  data: PluginStart
+  data: PluginStart,
+  alerting: AlertPluginStartContract
 ): ISearchStrategy<TimelineStrategyRequestType<T>, TimelineStrategyResponseType<T>> => {
+  const esAsInternal = data.search.searchAsInternalUser;
   const es = data.search.getSearchStrategy(ENHANCED_ES_SEARCH_STRATEGY);
+
   return {
     search: (request, options, deps) => {
-      if (request.factoryQueryType == null) {
+      const factoryQueryType = request.factoryQueryType;
+      const entityType = request.entityType;
+      const alertConsumers = request.alertConsumers;
+
+      if (factoryQueryType == null) {
         throw new Error('factoryQueryType is required');
       }
-      const queryFactory: TimelineFactory<T> = timelineFactory[request.factoryQueryType];
-      const dsl = queryFactory.buildDsl(request);
-      return es.search({ ...request, params: dsl }, options, deps).pipe(
-        map((response) => {
-          return {
-            ...response,
-            ...{
-              rawResponse: shimHitsTotal(response.rawResponse, options),
-            },
-          };
-        }),
-        mergeMap((esSearchRes) => queryFactory.parse(request, esSearchRes))
-      );
+
+      const queryFactory: TimelineFactory<T> = timelineFactory[factoryQueryType];
+
+      if (entityType != null && entityType === EntityType.ALERTS) {
```

Comment:
Is there a way to force `entityType` to be required on the request so we don't have to do these null checks?

Reply:
Yes! So my plan was to have a separate PR to update the front end code and that's when I would flip this to required. This allows me to not have to do it all in one PR>

Reply:
or you can do something simple like that `[EntityType.ALERTS].includes(entityType)`